galaxychop/galaxychop.py

```diff
@@ -15,7 +15,81 @@
 from scipy.interpolate import InterpolatedUnivariateSpline
 from sklearn.mixture import GaussianMixture
 import random
-        
+
+# #####################################################
+# GALAXY CLASS
+# #####################################################
+
+@attr.s(frozen=True)
+class Galaxy:
+    """This class builds a galaxy object from the masses, positions, and 
+    velocities of the particles (stars, dark matter, and gas).
+
+    Parameters
+    ----------
+    x_s, y_s, z_s: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Star positions
+    vx_s, vy_s, vz_s: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Star velocities
+    m_s: `np.ndarray(n,1)`
+        Star masses
+
+    x_dm, y_dm, z_dm: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Dark matter positions
+    vx_dm, vy_dm, vz_dm: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Dark matter velocities
+    m_dm: `np.ndarray(n,1)`
+        Dark matter masses    
+
+    x_g, y_g, z_g: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Gas positions
+    vx_g, vy_g, vz_g: `np.ndarray(n,1), np.ndarray(n,1), np.ndarray(n,1)`
+        Gas velocities
+    m_g: `np.ndarray(n,1)`
+        Gas masses 
+    
+    components_s: `np.ndarray(n_star,1)`
+        This indicates the component to which the stellar particle is assigned.
+        This is chosen as the most probable component.
+        `n_star` is the number of stellar particles.
+
+    components_g: `np.ndarray(n_gas,1)`
+        This indicates the component to which the gas particle is assigned.
+        This is chosen as the most probable component.
+        `n_gas` is the number of gas particles.
+
+    Atributes
+    ---------
+
+    """
+    
+    x_s = attr.ib()
+    y_s = attr.ib()
+    z_s = attr.ib()
+    vx_s = attr.ib()
+    vy_s = attr.ib()
+    vz_s = attr.ib()
+    m_s = attr.ib()
+    
+    x_dm = attr.ib()
+    y_dm = attr.ib()
+    z_dm = attr.ib()
+    vx_dm = attr.ib()
+    vy_dm = attr.ib()
+    vz_dm = attr.ib()
+    m_dm = attr.ib()
+
+    x_g = attr.ib()
+    y_g = attr.ib()
+    z_g = attr.ib()
+    vx_g = attr.ib()
+    vy_g = attr.ib()
+    vz_g = attr.ib()
+    m_g = attr.ib()
+
+    components_s = attr.ib(default=None)
+    components_g = attr.ib(default=None)
+    metadata = attr.ib()
 
```

Comment:
Does this class have an `__init__` method?

Reply:
I thought that with the decorator I was using there was no need to put it up. If it is necessary to add it, tell me and I will add it.

Reply:
That's right on spot. 
You don't need it. :100: